### PR TITLE
Update smart contract 2.0 design for state changes

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:
Update smart-contracts.md per latest HIP-260: Smart Contract Traceability

**Related issue(s)**:

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/design/smart-contracts.md [hunk 1]
```diff
@@ -373,20 +372,12 @@ Optional filters
 - `block_hash` should be retrieved by a join with the `record_file` table to find the `hash` of the file containing the
   transaction.
 - `hash` should be retrieved by a join with the `transaction` table
-- `hedera_child_transactions` (when added) will be retrieved by a join between the `contract_result` and transfer tables
-  (`assessed_custom_fee`, `crypto_transfer`, `token_transfer`, `nft_transfer`) tables based on child timestamps.
 - `logs` should be retrieved by a join between the `contract_result` and `contract_log` tables.
 - `state-changes` should be retrieved by a join between the `contract_result` and `contract_state_change` tables.
```
Comment (xin-hedera) [inline docs/design/smart-contracts.md:376]:
```suggestion
- `state_changes` should be retrieved by a join between the `contract_result` and `contract_state_change` tables.
```

**Review by xin-hedera — CHANGES_REQUESTED** [review]
just a nit

File: docs/design/smart-contracts.md (hunk 1 above)
Comment (steven-sheehy) [reply docs/design/smart-contracts.md]: Done

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Pull request merged**